feat: add support for output modifier flags

ruicalheno133 opened the pull request:
Currently, only `-e` option is used to execute an inline script. 

This PR:
- Adds support for osascript's output modifier flags `-s` (as specified [here](https://ss64.com/osx/osascript.html)). 
- Adds tests to ensure current behaviour is kept and `-s` flag behaviour works

I'm not sure if this solves #4, but using the `-s s` flag, it returns a string that can be parsed into an array. Neverthless, I think it is useful to have this option available to mirror the behaviour of the `osascript` command,.

Output without `-s` flag:
```
$ osascript -e 'return {{"foo", {"bar"}}}'        
foo, bar
```

Output with `-s` flag:
```
$ osascript -e 'return {{"foo", {"bar"}}}'  -s s
{{"foo", {"bar"}}}
```

ruicalheno133 committed:
add support for output modifier flags

sindresorhus commented:
The point of this package is to abstract away the command-line flags. I think this should instead be an option called `humanReadableReturnValue` that can be set to false to get the wanted behavior.

ruicalheno133 commented:
Got it 👍 Will update accordingly.

As a follow-up, I was wondering if it wouldn't be useful to also be able to read from a file. I could create a new function specifically for that or add a new argument `fromFile` and if this was set to true it would consider the `script` param as a filename. What do you think?

sindresorhus commented:
> As a follow-up, I was wondering if it wouldn't be useful to also be able to read from a file. I could create a new function specifically for that or add a new argument fromFile and if this was set to true it would consider the script param as a filename. What do you think?

I think the most common use-case for this package are short scripts you can just have inline. If you need to pass a file, you can simply call `fs.readFile`.

ruicalheno133 committed:
add humanReadableOutput argument to runAppleScript function

ruicalheno133 commented:
Right! Let's keep it simple. 

I added a new argument `humanReadableOutput` to the existing functions in 282e882. 
I thought wrapping it within an options object, but it seemed overkill since there are no more options to be defined.

sindresorhus commented:
> I thought wrapping it within an options object, but it seemed overkill since there are no more options to be defined.

I think it should be an options-object.

`runApplescript(..., false)` is not very readable.

sindresorhus commented:
https://github.com/sindresorhus/typescript-definition-style-guide#documentation

ruicalheno133 committed:
wrap humanReadableOutput in an options object

sindresorhus commented on index.d.ts:
````diff
@@ -14,14 +24,27 @@ const result = await runAppleScript('return "unicorn"');
+
+console.log(result);
+//=> '"unicorn"'
+```
````
The example should be on the docs for the `humanReadableOutput` option. Line 8.

ruicalheno133 replied:
Should I keep the example in both the functions and `humanReadableOutput` docs?

sindresorhus replied:
No, only in `humanReadableOutput`.

ruicalheno133 replied:
Done in cf16b68.

sindresorhus commented on index.d.ts:
````diff
@@ -14,14 +24,27 @@ const result = await runAppleScript('return "unicorn"');
+```
+import {runAppleScript} from 'run-applescript';
+
+const options = {
+    humanReadableOutput: false
+}
````
This can just be inline in the function call.

sindresorhus commented on index.js:
```diff
@@ -1,20 +1,28 @@
+	humanReadableOutput: true,
+};
+
+export async function runAppleScript(script, options = DEFAULT_OPTIONS) {
```
```suggestion
export async function runAppleScript(script, {humanReadableOutput = true} = {}) {
```

ruicalheno133 replied:
I believe the original approach ends up being more readable and easy to extend.
Nevertheless, I updated it as suggested in [da05ff8](https://github.com/sindresorhus/run-applescript/pull/14/commits/da05ff82061133e87977cfc2af0782a18c46356f), since I don't think there will be much more options to add.

sindresorhus commented on index.js:
```diff
@@ -1,20 +1,28 @@
 	}
 
-	const {stdout} = await execa('osascript', ['-e', script]);
+	const outputArg = options.humanReadableOutput ? [] : ['-ss'];
```
```suggestion
	const outputArguments = options.humanReadableOutput ? [] : ['-ss'];
```

ruicalheno133 replied:
I feel like outputArgs is expressive enough but updated it in [da05ff8](https://github.com/sindresorhus/run-applescript/pull/14/commits/da05ff82061133e87977cfc2af0782a18c46356f)

sindresorhus commented on test.js:
```diff
@@ -8,3 +8,19 @@ test('async', async t => {
 	t.is(runAppleScriptSync('return "unicorn"'), 'unicorn');
 });
+
+test('non human readable output async', async t => {
```
```suggestion
test('non-human readable output - async', async t => {
```

ruicalheno133 replied:
Fixed in [8ae5a3d](https://github.com/sindresorhus/run-applescript/pull/14/commits/8ae5a3dceab125fbda2be9ef95a515efbdfc40bc).

sindresorhus commented on index.d.ts:
```diff
@@ -1,7 +1,18 @@
+	/**
+	Modifies the output style.
+	If false returns value in recompilable source form.
+
```
Would be useful to link to the osascript docs about the different forms.

ruicalheno133 replied:
Agree. I added the link to osascript's man page in [da05ff8](https://github.com/sindresorhus/run-applescript/pull/14/commits/da05ff82061133e87977cfc2af0782a18c46356f)

sindresorhus commented:
Don't forget to add it to the readme. Use this formatting: https://github.com/sindresorhus/boxen#options

ruicalheno133 committed:
test: update test titles

ruicalheno133 committed:
refactor: turn options argument into inline object

- update documentation to accomodate new changes
- link osascript's man page to docs

sindresorhus committed:
Update index.d.ts

ruicalheno133 committed:
docs: update examples

sindresorhus commented:
> Don't forget to add it to the readme. Use this formatting: https://github.com/sindresorhus/boxen#options

ruicalheno133 committed:
docs: update api on readme.md

sindresorhus commented on readme.md:
````diff
@@ -11,7 +11,7 @@ $ npm install run-applescript
 
 ```js
-import {runAppleScript} from 'run-applescript';
+import { runAppleScript } from "run-applescript";
````
Don't make unrelated changes.

ruicalheno133 replied:
Sure. Linter automatically "messed" this up. 
Fixed it.

sindresorhus commented on readme.md:
```diff
@@ -21,14 +21,48 @@ console.log(result);
+
+##### humanReadableOutput
+
+Type: `boolean`
```
```suggestion
Type: `boolean`\
```

sindresorhus commented on readme.md:
```diff
@@ -21,14 +21,48 @@ console.log(result);
+Type: `boolean`
+Default: `true`
+
+Output modifier flag. When `false`, returns the value in a [recompilable source form](https://ss64.com/osx/osascript.html).
```
This is not the same as in index.d.ts

sindresorhus committed:
Update index.d.ts

sindresorhus committed:
Update test.js
